src/Symfony/Component/Scheduler/Trigger/PeriodicalTrigger.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Trigger;
+
+use Symfony\Component\Scheduler\Exception\InvalidArgumentException;
+
+/**
+ * @experimental
+ */
+class PeriodicalTrigger implements TriggerInterface, \Stringable
+{
+    private int $intervalInSeconds = 0;
+    private \DateTimeImmutable $from;
+    private \DateTimeImmutable $until;
+    private \DatePeriod $period;
+    private string $description = '';
+
+    public function __construct(
+        string|int|\DateInterval $interval,
+        string|\DateTimeImmutable $from = new \DateTimeImmutable(),
+        string|\DateTimeImmutable $until = new \DateTimeImmutable('3000-01-01'),
+    ) {
+        $this->from = \is_string($from) ? new \DateTimeImmutable($from) : $from;
+        $this->until = \is_string($until) ? new \DateTimeImmutable($until) : $until;
+
+        if (\is_int($interval)) {
+            self::ensurePositiveInteger($this->intervalInSeconds = $interval);
+
+            return;
+        }
+
+        if (\is_string($interval) && ctype_digit($interval)) {
+            self::ensureIntervalSize($interval);
+            $this->intervalInSeconds = (int) $interval;
+
+            return;
+        }
+
+        try {
+            if (\is_string($interval) && 'P' === ($interval[0] ?? '')) {
+                $this->intervalInSeconds = self::calcInterval($this->from, $this->from->add(new \DateInterval($interval)));
+
+                return;
+            }
+
+            $i = $interval;
+            if (\is_string($interval)) {
+                $this->description = $interval;
+                $i = \DateInterval::createFromDateString($interval);
+            } else {
+                $a = (array) $interval;
+                $this->description = \PHP_VERSION_ID >= 80200 && $a['from_string'] ? $a['date_string'] : 'DateInterval';
+            }
+            if ($this->canBeConvertedToSeconds($i)) {
+                $this->intervalInSeconds = self::calcInterval($this->from, $this->from->add($i));
+            } else {
+                $this->period = new \DatePeriod($this->from, $i, $this->until);
+            }
+        } catch (\Exception $e) {
+            throw new InvalidArgumentException(sprintf('Invalid interval "%s": ', $interval instanceof \DateInterval ? 'instance of \DateInterval' : $interval).$e->getMessage(), 0, $e);
+        }
+    }
+
+    public function __toString(): string
+    {
+        return $this->description ?: sprintf('every %d seconds', $this->intervalInSeconds);
```

Comment:
there is a case where description === 'DateInterval' but intervalInSeconds is true, we should then display the number of seconds instead of "DateInterval" I suppose?